packages/loader/container-loader/src/serializedStateManager.ts

```diff
@@ -61,104 +61,31 @@ export class SerializedStateManager {
 		}
 	}
 
-	private async getVersion(version: string | null): Promise<IVersion | undefined> {
-		const versions = await this.storageAdapter.getVersions(version, 1);
-		return versions[0];
-	}
-
 	public async fetchSnapshot(
 		specifiedVersion: string | undefined,
-		supportGetSnapshotApi: boolean | undefined,
+		supportGetSnapshotApi: boolean,
 	) {
-		const { snapshot, version } =
-			this.pendingLocalState === undefined
-				? await this.fetchSnapshotCore(specifiedVersion, supportGetSnapshotApi)
-				: { snapshot: this.pendingLocalState.baseSnapshot, version: undefined };
-		const snapshotTree: ISnapshotTree | undefined = isInstanceOfISnapshot(snapshot)
-			? snapshot.snapshotTree
-			: snapshot;
-		if (this.pendingLocalState) {
-			this.snapshot = {
-				tree: this.pendingLocalState.baseSnapshot,
-				blobs: this.pendingLocalState.snapshotBlobs,
-			};
-		} else {
+		if (this.pendingLocalState === undefined) {
+			const { snapshot, version } = supportGetSnapshotApi
+				? await fetchISnapshot(this.mc, this.storageAdapter, specifiedVersion)
+				: await fetchISnapshotTree(this.mc, this.storageAdapter, specifiedVersion);
+			const snapshotTree: ISnapshotTree | undefined = isInstanceOfISnapshot(snapshot)
+				? snapshot.snapshotTree
+				: snapshot;
 			assert(snapshotTree !== undefined, 0x8e4 /* Snapshot should exist */);
 			// non-interactive clients will not have any pending state we want to save
 			if (this.offlineLoadEnabled) {
 				const blobs = await getBlobContentsFromTree(snapshotTree, this.storageAdapter);
 				this.snapshot = { tree: snapshotTree, blobs };
 			}
+			return { snapshotTree, version };
+		} else {
+			this.snapshot = {
```

Comment:
nit, destructor the object
```suggestion
			const { baseSnapshot, snapshotBlobs} =this.pendingLocalState;
			this.snapshot = {
```

Reply:
could also make this.snapshot match pending local state in terms of property names.

Reply:
Good idea. I'll do that in another PR since it we will need to change those names in several places